vowpalwabbit/parse_args.cc

```diff
@@ -1424,7 +1424,7 @@ vw& parse_args(options_i& options, trace_message_t trace_listener, void* trace_c
 
 bool check_interaction_settings_collision(options_i& options, std::string file_options)
 {
-  bool command_line_has_interaction = options.was_supplied("q") || options.was_supplied("quadratic") ||
+  bool command_line_has_interaction = options.was_supplied("-q") || options.was_supplied("quadratic") ||
```

Comment:
This will explicitly not work for option providers such as json which do not use `-` or `--` like command lines do. This interface should operate on `q` or `quadratic` like it was before. The bug is in how this code handles the value of `-q`

Reply:
q was not registered with options object before calling this function.  Fixed it upstream to register -q and reverted this.

Reply:
Had to roll back registering quadratic since there are many other side effects that interact with loading the model.  Reverting back to testing for '-q' since quadratic option is not yet registered

Reply:
Ok changed was_supplied to look for '-' in addition to '--' instead of trying to register quadratic before calling this function.